airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/unstructured_parser.py

```diff
@@ -91,42 +115,119 @@ def parse_records(
                 }
 
     def _read_file(self, file_handle: IOBase, remote_file: RemoteFile, format: UnstructuredFormat, logger: logging.Logger) -> Optional[str]:
-        _import_unstructured()
-        if (
-            (not unstructured_partition_pdf)
-            or (not unstructured_partition_docx)
-            or (not unstructured_partition_pptx)
-            or (not unstructured_optional_decode)
-        ):
-            # check whether unstructured library is actually available for better error message and to ensure proper typing (can't be None after this point)
-            raise Exception("unstructured library is not available")
-
         filetype = self._get_filetype(file_handle, remote_file)
 
         if filetype == FileType.MD:
             file_content: bytes = file_handle.read()
-            decoded_content: str = unstructured_optional_decode(file_content)
+            decoded_content: str = optional_decode(file_content)
             return decoded_content
         if filetype not in self._supported_file_types():
-            self._handle_unprocessable_file(remote_file, format, logger)
+            self._handle_unprocessable_file(remote_file, format.skip_unprocessable_file_types, logger)
             return None
+        if format.processing.mode == "local":
+            return self._read_file_locally(file_handle, filetype, format.strategy)
+        elif format.processing.mode == "api":
+            try:
+                result: Optional[str] = self._read_file_remotely_with_retries(file_handle, format.processing, filetype, format.strategy)
+            except Exception as e:
+                # Re-throw as config error so the sync is stopped as problems with the external API need to be resolved by the user and are not considered part of the SLA.
+                # Once this parser leaves experimental stage, we should consider making this a system error instead for issues that might be transient.
+                raise AirbyteTracedException.from_exception(e)
```

Comment:
the default `failure_type` is FailureType.system_error so I don't think this will raise a config error. Might be worth verifying with a unit test

Reply:
That's a good point - I checked and this isn't stopping the sync because the file based stream implementation simply catches the exception and continues with the next file. There is no way of stopping the sync from within the parser right now - https://github.com/airbytehq/airbyte/pull/32589 will add it. I will turn this into a regular exception for now and revisit once https://github.com/airbytehq/airbyte/pull/32589 is merged